api/desecapi/tests/test_rrsets.py

```diff
@@ -42,15 +44,90 @@ def test_retrieve_my_rr_sets(self):
             self.assertStatus(response, status.HTTP_200_OK)
             self.assertEqual(len(response.data), 1, response.data)
 
+    def test_retrieve_my_rr_sets_pagination(self):
+        # Helper
+        def convert_links(links):
+            mapping = {}
+            for link in links.split(', '):
+                url, label = link.split('; ')
+                label = re.search('rel="(.*)"', label).group(1)
+                url = url[1:-1]
+                assert label not in mapping
+                mapping[label] = url
+            return mapping
+
+        # Prepare extra records so that we get three pages (total: n + 1)
+        n = int(settings.REST_FRAMEWORK['PAGE_SIZE'] * 2.5)
+        RRset.objects.bulk_create(
+            [RRset(domain=self.my_domain, subname=str(i), ttl=123, type='A') for i in range(n)]
+        )
+
+        # No pagination
```

Comment:
The following five blocks of code should be refactored into some sub-routine to make maintenance easier.

Reply:
done! looks much better now, thx for the suggestion